components/job-orchestration/job_orchestration/retention/streams_handler.py

```
@@ -0,0 +1,93 @@
+
+    # TODO: I feel it's ok to always run this even if there's no results from pymongo
+    # First, if we reached this line, it means either 1. no new targets has been added, or
+    # some new target has been added, and we have already removed them from mongodb. either case,
+    # the targets in the file must have been removed from the mongodb and not needed.
+    remove_targets(stream_output_config, targets_buffer.get_targets())
+    targets_buffer.flush()
+
+    return
+
+
+async def stream_retention(
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)